### DISCUSSION TITLE
Re Renders Component Page Several Times

### DISCUSSION THREAD

**Discussion by livevsonline** [TanStack/router#520] [General]
Hi

I am setting up a very simple router like this:

```
const testRoute= new Route({
	id: 'testRoute',
	getParentRoute: () => rootRoute,
	path: '/app/test/route',
	component: lazy(() => import("@/pages/test/testPage")),
});

```

My App.tsx looks like this:
```

import './App.css';
import {router} from "@/Routes";
import {RouterProvider} from "@tanstack/react-router";

function App() {
	return (
		<div className="App">
			<RouterProvider router={router}/>
		</div>
	)
}

export default App
```

In my testPage.tsx this is all i do:

```

function TestPage() {
	console.log('How Many Times Does Test Page Render');
	return <p>Hello Test Page</p>;
}

export default TestPage;
```

However the TestPage is rendered several times as demonstrated below:
![image](https://user-images.githubusercontent.com/39704356/217507543-7c374732-237f-4a2c-8f00-41a7882f4009.png)

Why is this happening? Is there a bug with the state flow of the router or is this normal/expected :)?

I'm on version: ^0.0.1-beta.61

**Comment by tannerlinsley** [comment]
Upgrade your version first. Then I can take a look.

Tanner Linsley

**Comment by esparter** [comment]
I am still seeing the same thing on latest version with component being rendered twice: "@tanstack/react-router": "^1.35.6",
I have simplified it down to a single root page:

```
import { createRootRoute } from "@tanstack/react-router";
export const Route = createRootRoute({
  component: Root,
});

function Root() {
  console.log("Render Root");
  return <div>Hello Root World!</div>;
}
```
I run "vite build" and serve with local static server to remove any dev issues.

<img width="388" alt="Screenshot 2024-06-12 at 15 51 56" src="https://github.com/TanStack/router/assets/53020154/2161f37c-6857-466a-9863-3ca02caa46f4">

**Reply by jacobwilson41** [reply]
me too on v1.36.3

**Reply by COUSCOUSZ** [reply]
same here

**Comment by ReeceHughes** [comment]
I'm also seeing double renders and am wondering if this is expected.  I was able to reproduce on the [Quickstart Example](https://tanstack.com/router/latest/docs/framework/react/examples/quickstart-file-based) on the website.  I added some `console.log`s in `index.tsx` and `about.tsx`.  Clicking on the links at the top of the page causes the component to render twice.

The example is using version 1.38.1.

![doubleRenderExample](https://github.com/TanStack/router/assets/21375570/33cd3720-1ec7-456b-8880-fef819037639)

**Reply by schiller-manuel** [reply]
can you please create a new github issue for this?

**Reply by ReeceHughes** [reply]
Yes, #1825.  Thanks for looking.

**Reply by imrankhanpersonalai** [reply]
same issue, did you fix it?
